### PR TITLE
Print NginxConfig in JSON format for debug and trace levels

### LINKED ISSUE #209: Print a JSON representation of the NginxConfig proto message in debug logs
At present, the `debug` and `trace` log levels have the ability to print out the `NginxConfig` protobuf message that is sent to the agent when a change is submitted:

https://github.com/nginx/agent/blob/cce39148ba5f027055d8eaac138221af2dc09318/src/plugins/nginx.go#L391

However, the string representation of such object is dictated by the `proto.CompactTextString()` function, that produces a serialized output that is hard to parse and read (specially for binary message fields), which makes debugging hard.

The proposed enhancement is to print a JSON-formatted version of the message, which would facilitate the debugging process.

### PR DESCRIPTION
**Pull request by dareste**
### Proposed changes

This PR changes the `log.traceF` and `log.debugF` function arguments in those cases where we log the `NginxConfig` protobuf message. Instead of the raw message (which is harder to parse and read when doing troubleshooting), the JSON version of the object is now logged.

Closes #209 

### Checklist

Before creating a PR, run through this checklist and mark each as complete.

- [x] I have read the [`CONTRIBUTING`](https://github.com/nginx/agent/blob/main/docs/CONTRIBUTING.md) document
- [ ] If applicable, I have added tests that prove my fix is effective or that my feature works
- [ ] If applicable, I have checked that any relevant tests pass after adding my changes
- [x] I have updated any relevant documentation ([`README.md`](https://github.com/nginx/agent/blob/main/README.md))

### CODE REVIEW

**Comment by dhurley** [comment]
Can you run `make deps` as well?

File: src/plugins/nginx.go [hunk 1]
```diff
@@ -388,7 +388,8 @@ func (n *Nginx) writeConfigAndReloadNginx(correlationId string, config *proto.Ng
 		return n.handleErrorStatus(status, message)
 	}
 
-	log.Debugf("WriteConfig start %v", config)
+	jsonConfig, _ := json.Marshal(config)
```
[the remark this answers is not in the document]
Comment (dareste) [reply src/plugins/nginx.go]: Agree, it's redundant. Line removed.

File: src/core/nginx.go [hunk 2]
```diff
@@ -316,7 +317,8 @@ func hasConfPath(files []*proto.File, confPath string) bool {
 }
 
 func (n *NginxBinaryType) WriteConfig(config *proto.NginxConfig) (*sdk.ConfigApply, error) {
-	log.Tracef("Writing config: %+v\n", config)
+	jsonConfig, _ := json.Marshal(config)
```
[the remark this answers is not in the document]
Comment (dareste) [reply src/core/nginx.go]: You're right, thanks for the suggestion!

**Comment by dareste** [comment]
@dhurley `make deps` is failing in my fork:

fatal: No names found, cannot describe anything.
cd sdk && go mod tidy && go mod vendor &&  make generate && go mod tidy && go mod vendor
go: finding module for package github.com/nginx/agent/sdk/v2/proto/events
go: finding module for package github.com/nginx/agent/sdk/v2/proto
github.com/nginx/agent/sdk/v2 imports
	github.com/nginx/agent/sdk/v2/proto: no matching versions for query "latest"
github.com/nginx/agent/sdk/v2/client imports
	github.com/nginx/agent/sdk/v2/proto/events: no matching versions for query "latest"
make: *** [deps] Error 1

File: src/core/nginx.go [hunk 3]
```diff
@@ -316,7 +317,11 @@ func hasConfPath(files []*proto.File, confPath string) bool {
 }
 
 func (n *NginxBinaryType) WriteConfig(config *proto.NginxConfig) (*sdk.ConfigApply, error) {
-	log.Tracef("Writing config: %+v\n", config)
+	if log.IsLevelEnabled(log.TraceLevel) {
+		jsonConfig, _ := json.Marshal(config)
```
Comment (Dean-Coakley) [inline src/core/nginx.go:321]: I think it make sense to handle/check the error returned by `json.Marshal`?

Comment (dareste) [reply src/core/nginx.go]: It makes sense, yes. I've changed the code to log the raw message if the marshal operation fails.

**Review by oliveromahony — APPROVED** [review]

**Comment by oliveromahony** [comment]
make deps looks outdated, I'll do a separate PR for that

**Review by Dean-Coakley — APPROVED** [review]
